qiita_db/support_files/patches/python_patches/14.py

```diff
@@ -0,0 +1,20 @@
+# Feberuary 7, 2015
+# This patch recreates all the QIIME mapping files to avoid lower-upper case
+# problems. See https://github.com/biocore/qiita/issues/799
+
+from qiita_db.util import get_mountpoint
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+_id, fp_base = get_mountpoint('templates')[0]
+
+for prep_template_id in conn_handler.execute_fetchall(
+        "SELECT prep_template_id FROM qiita.prep_template"):
+    prep_template_id = prep_template_id[0]
+    pt = PrepTemplate(prep_template_id)
+    study_id = pt.study_id
+
+    for _, fpt in pt.get_filepaths():
+        pt.create_qiime_mapping_file(fpt)
```

Comment:
Good question. I think that theoretically, it should be okay to do it this way, but I'm slightly hesitant because I'm not sure if there are places in the code that unadvisable assumptions are made with regard to the number of templates associated with a study (e.g., is there code that assumes that there is only one prep template, and if so, does that code presume to use the prep template with the highest ID, or the lowest ID?)